File: scala/src/main/scala/folio/codinginterview/domain/StockSymbol.scala
package folio.codinginterview.domain

/** 銘柄を表す。 */
enum StockSymbol {
  case Toyopa, Somy
}

object StockSymbol {
  def fromString(s: String): Option[StockSymbol] = s match {
    case "Toyopa" => Some(Toyopa)
    case "Somy"   => Some(Somy)
    case _        => None
  }
}

File: php/src/CodingInterview/Domain/StockSymbol.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Domain;

/** 銘柄を表す。 */
enum StockSymbol: string
{
    case Toyopa = 'Toyopa';
    case Somy = 'Somy';

    public static function fromStringOrNull(string $s): ?StockSymbol
    {
        return match ($s) {
            'Toyopa' => self::Toyopa,
            'Somy' => self::Somy,
            default => null,
        };
    }
}
